tests/unit/test_charm_fiveg_n2_relation_joined.py

```diff
@@ -0,0 +1,215 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+from unittest.mock import PropertyMock, patch
+
+import pytest
+import scenario
+from ops.pebble import Layer, ServiceStatus
+
+from charm import AMFOperatorCharm
+from k8s_service import K8sService
+
+NRF_URL = "http://nrf:8081"
+
+
+class TestCharmFiveGN2RelationJoined:
+    patcher_check_output = patch("charm.check_output")
+    patcher_k8s_service = patch("charm.K8sService", autospec=K8sService)
+    patcher_nrf_url = patch(
+        "charms.sdcore_nrf_k8s.v0.fiveg_nrf.NRFRequires.nrf_url", new_callable=PropertyMock
+    )
+    patcher_is_resource_created = patch(
+        "charms.data_platform_libs.v0.data_interfaces.DatabaseRequires.is_resource_created"
+    )
+
+    @pytest.fixture(autouse=True)
+    def context(self):
+        self.ctx = scenario.Context(
+            charm_type=AMFOperatorCharm,
+        )
+
+    @pytest.fixture(autouse=True)
+    def setup(self):
+        self.mock_is_resource_created = (
+            TestCharmFiveGN2RelationJoined.patcher_is_resource_created.start()
```

Comment:
since the patches are being manually started (without using decorators or with statements) they also need to be stopped.

Reply:
Done